### PR TITLE
As/phonenumber format

### PR DESCRIPTION
**Pull request by magnearun**
Format phonenumber input field

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/parental-leave/src/forms/ParentalLeaveForm.ts [hunk 1]
```diff
@@ -94,6 +94,7 @@ export const ParentalLeaveForm: Form = buildForm({
                   name: 'Símanúmer',
                   id: 'applicant.phoneNumber',
                   variant: 'tel',
+                  placeholder: '000-00000',
```
Comment (arnifreyrs) [inline libs/application/templates/parental-leave/src/forms/ParentalLeaveForm.ts:97]: too many zeros

**Review by arnifreyrs — APPROVED** [review]
Awesome!

File: libs/shared/form-fields/src/lib/InputController/InputController.tsx [hunk 2]
```diff
@@ -26,9 +29,37 @@ export const InputController: FC<Props> = ({
...
+        name={name}
+        render={({ value, onChange }) => (
+          <NumberFormat
+            customInput={Input}
+            id={id}
+            disabled={disabled}
+            name={name}
+            placeholder={placeholder}
+            label={label}
+            type={type as 'text' | 'tel' | 'password'}
```
Comment (jeremybarbet) [inline libs/shared/form-fields/src/lib/InputController/InputController.tsx:49]: I'm curious to know why the password has to use the numberformat field?

Comment (oddsson) [reply libs/shared/form-fields/src/lib/InputController/InputController.tsx:49]: ☝️

**Review by oddsson — COMMENTED** [review]
Why use type password for number fields?

File: libs/shared/form-fields/src/lib/InputController/InputController.tsx (hunk 2 above)
Comment (magnearun) [reply libs/shared/form-fields/src/lib/InputController/InputController.tsx]: @jeremybarbet , @oddsson yeah this does not need to be there, I just copied the type from the NumberFormat component. Removed it.

File: libs/shared/form-fields/src/lib/InputController/InputController.tsx [hunk 3]
```diff
@@ -25,10 +29,68 @@ export const InputController: FC<Props> = ({
...
+            }}
+            hasError={error !== undefined}
+            errorMessage={error}
+          />
+        )}
+      />
+    )
+  }
+
+  if (format && ['text', 'tel'].includes(type)) {
```
Comment (arnifreyrs) [inline libs/shared/form-fields/src/lib/InputController/InputController.tsx:68]: hmm shouldn't `number` also be in the array? Like bank info is essentially of type number and has a format `####-##-######`

**Review by arnifreyrs — APPROVED** [review]
great stuff

**Pull request merged**